hackathon_site/dashboard/frontend/src/api/types.ts

```diff
@@ -42,3 +42,45 @@ export interface Category {
     max_per_team: number;
     unique_hardware_count: number;
 }
+
+/** Event API */
+export interface Profile {
+    id: number;
+    id_provided: boolean;
+    attended: boolean;
+    acknowledge_rules: boolean;
+    e_signature: string;
+    team: number;
+}
+
+interface UserWithoutProfile {
+    id: number;
+    first_name: string;
+    last_name: string;
+    email: string;
+}
+
+export interface User extends UserWithoutProfile {
+    profile: Profile;
+}
+
+export interface Team {
+    id: number;
+    team_code: string;
+    created_at: Date;
+    updated_at: Date;
+    profiles: Profile & { user: UserWithoutProfile }[];
```

Comment:
Great use of typescript magic! Two things:

First, this needs an extra set of brackets to work properly `(Profile & { user: UserWithoutProfile })[]`

Second, from looking at #324, the nested `profile` don't have a `team` property which this currently expects. You can use the [`Omit`](https://www.typescriptlang.org/docs/handbook/utility-types.html#omittype-keys) utility type to remove it from `Profile`, but that might start to get messy.

Reply:
Okay so what I did was omit the team property from Profile, and from the UserWithoutProfile. The changes are in the new commit, I'm not too sure how to clean it up or if it works.